spacepy/datamodel.py

```diff
@@ -1337,12 +1337,23 @@ def readJSONMetadata(fname, **kwargs):
     if hasattr(fname, 'read'):
         lines = fname.read()
     else:
-        with open(fname, 'r') as f:
-            lines = f.read()
+        # also possible in an exploration sense utilizing UnicodeDecodeError
```

Comment:
Will this really let us guess on a gzip file? In loadpickle we just sort of try gzip as a long shot if the read fails for any reason.

Reply:
THat was the thinking was to try and guess, but just `*.gz` is good by me and how it works now.

Reply:
Updated